core/src/main/java/io/aiven/kafka/tieredstorage/RemoteStorageManager.java

```diff
@@ -265,7 +292,8 @@ boolean requiresCompression(final LogSegmentData logSegmentData) {
     }
 
     private void uploadSegmentLog(final RemoteLogSegmentMetadata remoteLogSegmentMetadata,
-                                  final TransformFinisher transformFinisher)
+                                  final TransformFinisher transformFinisher,
+                                  final SegmentCustomMetadataBuilder customMetadataBuilder)
```

Comment:
Here and in other places: should we add more arguments or would it better to return the size instead?

Reply:
There may be some refactoring possible, though I'd suggest doing it in a separate PR.